### PR TITLE
feat: Add withCapacity static method. Use protected instead private access modifiers

### PR DESCRIPTION
**Pull request by MaxGraey**
Alternative approach of #4

### CODE REVIEW

**Comment by yjhmelody** [comment]
This cannot completely replace ensureCapacity, because this function is called only once. There is obviously a `shrink` method, but why not add a corresponding one? I can’t understand what practical application scenarios `shrink` can do. `ensureCapacity` is more commonly used than `shrink` IMO.

**Comment by yjhmelody** [comment]
Could you change the `private` field to `protected` by the way? Maybe I need to inherit it.

**Comment by MaxGraey** [comment]
done

File: assembly/index.ts [hunk 1]
```diff
@@ -10,10 +10,14 @@ export class StringSink {
   private buffer: ArrayBuffer;
   private offset: u32 = 0;
```
Comment (yjhmelody) [inline assembly/index.ts]: I also mean:
```ts
protected buffer: ArrayBuffer;
protected offset: u32 = 0;
```

**Pull request merged**